src/vs/workbench/services/workspaces/common/workspaceTrust.ts

```diff
@@ -129,7 +133,8 @@ export class WorkspaceTrustManagementService extends Disposable implements IWork
 		}
 
 		if (this.workspaceService.getWorkbenchState() === WorkbenchState.EMPTY) {
-			return true;
+			// Check memento, otherwise default to the setting value
+			return this._memento.isTrusted ?? this.configurationService.inspect<boolean>(WORKSPACE_TRUST_EMPTY_WORKSPACE).userValue ?? false;
```

Comment:
I had to push a fix for the enabled setting as well. This will fail if we change the default so you should fallback to 
```ts
this.configurationService.inspect<boolean>(WORKSPACE_TRUST_EMPTY_WORKSPACE).defaultValue
```
if the userValue is not defined

Reply:
Given that the setting is defined with the `Application` scope, I wonder if we can use `getValue()` instead of `inspect`.

Reply:
i think you are right, this was not possible when the setting was hidden, that's why it was this way for the enablement case.